### ISSUE TITLE
Spurious Unknown with typing.Type

### ISSUE THREAD

**Issue by mikeshardmind** [microsoft/pyright#4988]
**Describe the bug**

pyright is adding unknown to parts of a fully typed function signature, seemingly replacing `Type` with `Unknown`

**To Reproduce**

```
from msgspec.toml import decode
```

**Expected behavior**

This shouldn't treat Type as Unknown

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

```py
from msgspec.toml import decode
```

leads to this error:
```
Type of "decode" is partially unknown
  Type of "decode" is "Overload[(buf: bytes | str, *, dec_hook: ((Unknown, Any) -> Any) | None = None) -> Any, (buf: bytes | str, *, type: Type[T@decode] = ..., dec_hook: ((Unknown, Any) -> Any) | None = None) -> T@decode, (buf: bytes | str, *, type: Any = ..., dec_hook: ((Unknown, Any) -> Any) | None = None) -> Any, (buf: Unknown, *, type: Any = Any, dec_hook: Unknown | None = None) -> (dict[str, Any] | Unknown | Any)]"
```

msgspec==0.14.2
Definition of the fully typed method involved is here (linking to as it appears in the version installed): https://github.com/jcrist/msgspec/blob/f5d7d17932560f7b0f9fb8eb45a73d3d601f97d1/msgspec/toml.py#L87-L114
```py
@overload
def decode(
    buf: Union[bytes, str], *, dec_hook: Optional[Callable[[Type, Any], Any]] = None
) -> Any:
    pass


@overload
def decode(
    buf: Union[bytes, str],
    *,
    type: Type[T] = ...,
    dec_hook: Optional[Callable[[Type, Any], Any]] = None,
) -> T:
    pass


@overload
def decode(
    buf: Union[bytes, str],
    *,
    type: Any = ...,
    dec_hook: Optional[Callable[[Type, Any], Any]] = None,
) -> Any:
    pass


def decode(buf, *, type=Any, dec_hook=None):
```

**VS Code extension or command-line**
As part of pylance as a language server in visual studio code, pylance version: v2023.4.30

**Additional context**

Only an error with strict type checking or with `reportUnknownMemberType` specifically enabled ontop of basic

**Comment by erictraut** [comment]
This library is using `typing.Type`, which is a generic class that accepts a single type argument. However, no type argument is provided, so a type checker will assume `Type[Any]` (or `Type[Unknown]`, which is an implicit `Any`). If the intent of the library is to accept any `Type`, it should specify `Type[Any]` or use the lower-case builtin `type`.
